src/tellor_disputables/data.py

```diff
@@ -109,19 +109,28 @@ async def log_loop(web3: Web3, addr: str) -> list[tuple[int, Any]]:
     return unique_events_list
 
 
-async def is_disputable(reported_val: float, query_id: str, conf_threshold: float) -> Optional[bool]:
+async def general_fetch_new_datapoint(feed: DataFeed) -> Optional[Any]:
```

Comment:
what purpose does this function serve?

Reply:
it's so I can mock the failure of fetching a new datapoint easier in tests

Reply:
oh gotcha sounds good